static/js/zamboni/static_theme.js

```diff
@@ -0,0 +1,244 @@
+            set: function(color) {
+                $('text.textcolor').attr('fill', color);
+            }
+        });
+
+        Object.defineProperty(this, 'toolbar', {
+            set: function(color) {
+                var attrs;
+                if (!color) {
+                    attrs = {'fill': '#fff', 'fill-opacity': "0.6"};
+                } else {
+                    attrs = {'fill': color, 'fill-opacity': "1"};
+                }
+                $('rect.toolbar').attr(attrs);
+            }
+        });
+
+        Object.defineProperty(this, 'toolbar_text', {
+            set: function(color) {
+                $('text.toolbar_text').attr('fill', color);
+            }
+        });
+
+        Object.defineProperty(this, 'toolbar_field', {
+            set: function(color) {
+                if (!color) {
+                    color = '#fff';
+                }
+                $('rect.toolbar_field').attr('fill', color);
+            }
+        });
+
+        Object.defineProperty(this, 'toolbar_field_text', {
+            set: function(color) {
+                $('text.toolbar_field_text').attr('fill', color);
+            }
+        });
+
+        this.updateHeaderURL = function(src, width) {
+            var header_img = $('#svg-header-img');
+
+            var img  = new window.Image();
+            $(img).on('load', function() {
+                header_img.attr({width: img.width, height: img.height});
+            });
+            header_img.attr('href', (img.src = header_img.src = src));
+
+            var div = header_img.parents('div')[0];
+            header_img.attr('x', div.clientWidth - width);
+        };
+    }
+
+
+    $('#theme-wizard').each(initThemeWizard);
+
+
+    function initThemeWizard() {
+        var $wizard = $(this),
+            browserPreview = new StaticThemeBrowserPreview();
+
+        browserPreview.generate($('#browser-preview'));
+
+        $wizard.on('click', '.reset', _pd(function() {
+            var $this = $(this),
+            $row = $this.closest('.row');
+            $row.find('input[type="hidden"]').val('');
+            $row.find('input[type=file], .note').show();
+            var preview = $row.find('.preview');
+            preview.removeAttr('src').removeClass('loaded');
+            $this.hide();
+        }));
+
+        $wizard.on('change', 'input[type="file"]', function() {
+            var $row = $(this).closest('.row');
+            var reader = new FileReader(),
+                file = getFile($row.find('input[type=file]'));
+            if (!file) return;  // don't do anything if no file selected.
+            $row.find('input[type=file], .note').hide();
+
+            reader.onload = (function(aImg) { return function(e) {
+                aImg.attr('src', e.target.result);
+                aImg.show().addClass('loaded');
+                $row.find('.reset').show();
+                updateManifest();
+            };})($row.find('.preview'));
```

Comment:
This is a bit hard to read... Maybe just having the `aImg` in the parent function or using `.bind()` or similar would be nicer

Reply:
https://github.com/mozilla/addons-server/pull/6993/commits/da0ad769443c0fda48457a9f857bbf89fc0e24af